snapcraft.yaml

```diff
@@ -93,21 +93,52 @@ parts:
     plugin: nil
     build-packages:
       - wget
-    override-pull: |
-      # Do not use rustup to work around https://forum.snapcraft.io/t/armhf-builds-on-launchpad-timing-out/31008
+    override-pull: wget -O rustup.sh https://sh.rustup.rs
+    override-build: |
       REQUIRED_RUST_VERSION=1.78.0
-      ROOT=https://static.rust-lang.org/dist/rust-$REQUIRED_RUST_VERSION
-      if [ $CRAFT_ARCH_BUILD_FOR = "amd64" ]; then
-        BINARIES_SUFFIX=x86_64-unknown-linux-gnu
-      elif [ $CRAFT_ARCH_BUILD_FOR = "armhf" ]; then
-        BINARIES_SUFFIX=armv7-unknown-linux-gnueabihf
-      elif [ $CRAFT_ARCH_BUILD_FOR = "arm64" ]; then
-        BINARIES_SUFFIX=aarch64-unknown-linux-gnu
+      sh rustup.sh -q -y --default-toolchain $REQUIRED_RUST_VERSION
+      if [ $CRAFT_ARCH_BUILD_ON != $CRAFT_ARCH_BUILD_FOR ]; then
+        if [ $CRAFT_ARCH_BUILD_FOR = "arm64" ]; then
+          $HOME/.cargo/bin/rustup target add aarch64-unknown-linux-gnu
+        elif [ $CRAFT_ARCH_BUILD_FOR = "armhf" ]; then
+          $HOME/.cargo/bin/rustup target add thumbv7neon-unknown-linux-gnueabihf
+        fi
       fi
-      wget -O - $ROOT-$BINARIES_SUFFIX.tar.gz | tar -x -z --strip-components=1
-      ./install.sh --prefix=/usr --destdir=$CRAFT_STAGE
     override-prime: ''
 
+  # For thunderbird beta we refer to mozilla-beta trees on fetching the
```

Comment:
I'm not sure to understand that comment, there is no `mozilla-beta` mention in the snapcraft.yaml?

Reply:
`candidates` here is for beta, and the point is that we pick information from mozilla-beta tree's content (via `FETCHES`) rather than `comm-beta` (Thunderbird)